### PR TITLE
Refactors cable and disposal pipe construction/deconstruction code to improve performance

### PR DESCRIPTION
**Pull request by park66665**
## About The Pull Request

(In case you are curious, my branch names are selected from what I was listening when I created the branch. Maxim Vengerov 1992 recording, it was.)

1. Every cable structure creates and stores a cable coil item since its creation...
1.1. when it is only needed for destruct item drop.
1.2 And every disposal pipe structure also does the same thing with disposalconstruct structure!
2. So, do you know how many cables and disposal pipes are there in a map?
3. And, do you know atmosmachineries already create broken parts in situ? (since tgstation/tgstation#21992 thx MSO)
4. Therefore, this PR fixes the problem by making them create post-destruction objects in situ too, like atmosmachineries!
5. Additionally, the PR ended up making the code simpler.

Finally  a downstream note: AU should revert austation/austation#2311 and make changes to austation/austation#2028 to mirror this PR.

## Why It's Good For The Game

To summarize, this should shorten the world init time (around 1%), and reduce memory usage a little: around 10MB. ~~why do you use so much memory on these, BYOND~~

## Changelog
:cl:
refactor: refactored cable and disposal pipe construction/deconstruction code to remove ugly legacy code
fix: removal of said ugly legacy code improved performance a bit
fix: and fixed cable color bugs
/:cl:

### CODE REVIEW

**Comment by utilbird** [comment]
PR description is confusing, that's now how you use lists xd

**Comment by park66665** [comment]
To respond to your request, I maximized transition words. Therefore, I want to ask you if the PR description is now thoroughly beginner-friendly.

**Comment by utilbird** [comment]
> PR description is now thoroughly beginner-friendly.

rude, I understood what it meant but it wasn't an intuitive way to explain it :pensive:

**Comment by LemonInTheDark** [comment]
Kinda based ngl

File: code/modules/recycling/disposal/bin.dm [hunk 1]
```diff
@@ -234,16 +229,10 @@
...
-		if(stored)
-			stored.forceMove(T)
-			src.transfer_fingerprints_to(stored)
-			stored.anchored = FALSE
-			stored.density = TRUE
-			stored.update_icon()
+		new /obj/structure/disposalconstruct(loc, null, SOUTH, FALSE, src)
 	for(var/atom/movable/AM in src) //out, out, darned crowbar!
-		AM.forceMove(T)
+		AM.forceMove(loc)
```
Comment (utilbird) [inline code/modules/recycling/disposal/bin.dm]: `get_turf(src)` is the standard for content emptying forceMoves, although this will work 99.9% of the time

**Comment by francinum** [comment]
Interesting note: Windoors also do this, Storing like 4 items out of bounds to only very rarely ever be utilized.

**Comment by ike709** [comment]
Xeno said it's ready and I trust them enough to test their code. Let's see if this backfires.
